### PR TITLE
build: Add symlink creation to container image manipulation API.

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

File: meta-mender-qemu/scripts/docker/ext4_manipulator.py [hunk 1]
```diff
@@ -35,6 +35,21 @@ def put(local_path, remote_path, rootfs, remote_path_mkdir_p=False):
...
+    proc = subprocess.Popen(["debugfs", "-w", rootfs], stdin=subprocess.PIPE,
+                            stderr=subprocess.STDOUT)
+    if remote_path_mkdir_p:
+        # Create parent directories sequencially, to simulate a "mkdir -p" on the final dir
+        parent_dirs = list(PurePath(remote_path).parents)[::-1][1:]
+        for parent in parent_dirs:
+            proc.stdin.write(("mkdir %s\n" % parent).encode())
+    proc.stdin.write(("cd %s\n" % os.path.dirname(remote_path)).encode())
+    proc.stdin.write(("rm %s\n" % os.path.basename(remote_path)).encode())
+    proc.stdin.write(("symlink %s %s\n" % (os.path.basename(remote_path), target)).encode())
```
Comment (lluiscampos) [inline meta-mender-qemu/scripts/docker/ext4_manipulator.py:48]: Except for this line, everything else is the same as `put`... consider refactoring :wink:

**Review by lluiscampos — APPROVED** [review]

File: meta-mender-qemu/scripts/docker/ext4_manipulator.py (hunk 1 above)
Comment (kacf) [reply meta-mender-qemu/scripts/docker/ext4_manipulator.py:48]: Fair point!

**Comment by kacf** [comment]
Note that I have submitted two different jobs for this:
* [Pure build](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1468498786) - If this is green, this PR can be merged.
* [Test with](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1468502847) https://github.com/mendersoftware/mender/pull/1667 - This may be red for unrelated reasons, and should not count.

**Review by lluiscampos — APPROVED** [review]

**Comment by lluiscampos** [comment]
> Note that I have submitted two different jobs for this:
> 
>     * [Pure build](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1468498786) - If this is green, this PR can be merged.
> 

Passed. Merging this PR.

>     * [Test with](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1468502847) [Fixes for `--stop-before` flag. mender#1667](https://github.com/mendersoftware/mender/pull/1667) - This may be red for unrelated reasons, and should not count.

Failed. But as you said it is to be followed-up separately.

**Pull request merged**